sdk/core/azure-core/src/main/java/com/azure/core/util/RequestContent.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.implementation.util.ArrayContent;
+import com.azure.core.util.serializer.ObjectSerializer;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+/**
+ * Represents the content sent as part of a request.
+ */
+public abstract class RequestContent {
+    /**
+     * Write the {@link RequestContent} to the {@link RequestOutbound}.
+     *
+     * @param requestOutbound The outbound where the request will be written.
+     * @return An asynchronous response which will emit once the request has completed writing.
+     */
+    public abstract Mono<Void> writeToAsync(RequestOutbound requestOutbound);
+
+    /**
+     * Write the {@link RequestContent} to the {@link RequestOutbound}.
+     *
+     * @param requestOutbound The outbound where the request will be written.
+     */
+    public abstract void writeTo(RequestOutbound requestOutbound);
+
+    /**
+     * Gets the length of the {@link RequestContent}, if it is able to be calculated.
+     *
+     * @return The length of the {@link RequestContent}, if it is able to be calculated.
+     */
+    public abstract Long attemptToGetLength();
```

Comment:
I would just call this `getLength`, with the documentation clearly stating this is a best-effort only and may return -1 or some other value indicating that the value is unable to be calculated.